showcase/README.md

```diff
@@ -0,0 +1,49 @@
+# Showcase integration testing
+
+This submodule contains the gapic-showcase integration tests for
+gapic-generator-go, as well as the script used to setup and execute them.
+
+## How the tests work
+
+The tests can be run out-of-the-box in the normal `go test` fashion, so long as
+there is a `gapic-showcase` server already running locally. This does not test
+the a client generated with the locally checked out generator - it would use the
+released version of the GAPIC. Running `make test` will execute the tests
+against the locally install generator.
+
+To test the local generator, the `showcase.bash` script first downloads the
+Showcase artifacts associated with the targeted version. These are the compiled
+proto descriptor set, the retry configuration, and a pre-compiled server binary.
+
+Then using protoc and the retrieved artifacts as input, the Go protobuf/gRPC
+bindings, and the Go GAPIC are generated, the latter with the
+**locally installed** gapic-generator-go. Make sure to `make install` so that
+any new changes are utilized during generation.
+
+This submodule's `go.mod` is temporarily edited to replace the remote dependency
+on `gapic-showcase` with the locally generated artifacts.
+
+The server binary is run in the background and the tests in this directory are
+executed against it.
+
+### Adding tests
+
+Adding tests for an existing service is as easy as adding a new Go test to the
+respective file: `echo_test.go`, `identity_test.go`, or `sequence_test.go`.
+
+Adding a new service for testiing requires creating an appropriate
+`{service}_test.go` file and adding client creation to the `main_test.go`.
+
+Any newly added tests should make sure to add a `leakcheck` statement to the
+test function: `defer check(t)`.
+
+### Updating the Showcase version
```

Comment:
This is probably not worth adding in this doc, but I'll bring it up since it's on my mind: to use a local version of Showcase, one would also need to do a replace in go.mod to point to the local repo.

Reply:
This is not easy to do atm. This script depends on the compiled proto descriptor, which is created by the release process. Locally, you'd have to compile that descriptor, then we'd have to wire up the script to check if some environment variable is set to point it at the locally compiled descriptor (or you'd just copy it into this dir and comment out the `curl`).

Using a local version of showcase for any of this is not set up to be easy atm. But that can change!